### PR TITLE
allow comma in add, also tests

### PR DESCRIPTION
**Pull request by Moelf**
Fix https://github.com/JuliaLang/Pkg.jl/issues/948

(this is much harder than I thought, and maybe I'm still doing something wrong but all tests passed)

### CODE REVIEW

**Comment by fredrikekre** [comment]
bors try

**Comment by fredrikekre** [comment]
While this seems to work I think we need to be a bit more careful and only replace `','` with `' '` when we expect a list of package arguments. For example, I think that
```
pkg> status, Example
```
should still error, and everything inside a quoted string should not be touched.

**Comment by Moelf** [comment]
I agree. Then it's a matter of where to patch this, we can add a `if startswith(` at the place, or we can play with `do_add!()`?

**Comment by Moelf** [comment]
I added a section to `lex` to handle commas only when it's one of `add|rm|remove` && `,` is outside of quotes, tests seem passing. @fredrikekre take a look whenever, thx!

**Comment by Moelf** [comment]
bors try

**Comment by fredrikekre** [comment]
bors try

File: src/REPLMode.jl [hunk 1]
```diff
@@ -208,6 +209,14 @@ function lex(cmd::String)::Vector{QString}
                 push_token!(false)
                 push!(qstrings, QString(";", false))
             end
+        elseif c == ','
+            if in_doublequote || in_singlequote || !replace_comma# raw char
```
Comment (StefanKarpinski) [inline src/REPLMode.jl]:
```suggestion
            if in_doublequote || in_singlequote || !replace_comma # raw char
```
I don't really follow what this comment means...

Comment (Moelf) [reply src/REPLMode.jl]: I'm not sure but it appears several time in the lexer
https://github.com/JuliaLang/Pkg.jl/blob/6d8f0727c8809e4b06da29acda8b55e1cfa4dcb1/src/REPLMode.jl#L199

Comment (StefanKarpinski) [reply src/REPLMode.jl]: Ok, I guess carry on then. @00vareladavid, can you explain what the comment means?

Comment (00vareladavid) [reply src/REPLMode.jl]: Yeah it means "not a delimiter" i.e. a character that will eventually be used as part of some identifier. Perhaps that should be more clear

**Comment by 00vareladavid** [comment]
@Moelf have you checked to see how well this plays with autocompletions? I took a quick shot look at adding commas a while back and I remember autocompletions being a problem. IIRC I took a different approach, so it might not be a problem at all with your implementation

**Comment by Moelf** [comment]
Seems, working:
```
(Pkg) pkg> add Exam <Tab> ---> add Example
(Pkg) pkg> add Example, HTT <Tab> ---> add Example, HTTP
```

`remove` has the same behavior.

**Comment by Moelf** [comment]
Can this be merged?

**Pull request merged**